examples/engine/1_basics.py

```diff
@@ -1,46 +1,64 @@
 
-from dff.core.pipeline import Pipeline
-from dff.utils.testing.common import check_happy_path, is_interactive_mode, run_interactive_mode
+from dff.utils.testing.common import (
+    check_happy_path,
+    is_interactive_mode,
+    run_interactive_mode,
+)
+
 
-# First of all, to create a dialog agent, we need to create a dialog script.
-# Below, `script` is the dialog script.
-# A dialog script is a flow dictionary that can contain multiple script .
-# Script are needed in order to divide a dialog into sub-dialogs and process them separately.
-# For example, the separation can be tied to the topic of the dialog.
-# In our example, there is one flow called greeting_flow.
+# %% [markdown]
+"""
+First of all, to create a dialog agent, we need to create a dialog script.
+Below script means a dialog script.
+A script is a dictionary, where the keys are the names of the flows.
+A script can contain multiple scripts, what is needed in order to divide
+a dialog into sub-dialogs and process them separately.
+For example, the separation can be tied to the topic of the dialog.
+In this example there is one flow called `greeting_flow`.
+
+Flow describes a sub-dialog using linked nodes.
+Each node has the keywords `RESPONSE` and `TRANSITIONS`.
 
-# Inside each flow, we can describe a sub-dialog.
-# Here we can also use keyword `LOCAL`, which we have considered in other examples.
+* `RESPONSE` contains the response
+    that the agent will return from the current node.
+* `TRANSITIONS` describes transitions from the
+current node to another nodes. This is a dictionary,
+where keys are names of the nodes and
+values are conditions of transition to them.
+"""
 
-# Flow describes a sub-dialog using linked nodes, each node has the keywords `RESPONSE` and `TRANSITIONS`.
 
-# `RESPONSE` - contains the response that the dialog agent will return when transitioning to this node.
-# `TRANSITIONS` - describes transitions from the current node to other nodes.
-# `TRANSITIONS` are described in pairs:
-#      - the node to which the agent will perform the transition
-#      - the condition under which to make the transition
+# %%
 toy_script = {
     "greeting_flow": {
-        "start_node": {  # This is an initial node, it doesn't need an `RESPONSE`
+        "start_node": {  # This is the initial node,
+            # it doesn't contain a `RESPONSE`.
             RESPONSE: "",
-            TRANSITIONS: {"node1": cnd.exact_match("Hi")},  # If "Hi" == request of user then we make the transition
+            TRANSITIONS: {"node1": cnd.exact_match("Hi")},
+            # If "Hi" == request of the user then we make the transition.
         },
         "node1": {
-            RESPONSE: "Hi, how are you?",  # When the agent goes to node1, we return "Hi, how are you?"
-            TRANSITIONS: {"node2": cnd.exact_match("i'm fine, how are you?")},
+            RESPONSE: "Hi, how are you?",  # When the agent goes
+            # to node1 we return "Hi, how are you?".
```

Comment:
```suggestion
            RESPONSE: "Hi, how are you?",  # When the agent enters node1,
            # return "Hi, how are you?".
```

Reply:
Is the agent really in the node and "enter" better?

Reply:
Yes, while awaiting a response from user, the agent is being inside a certain node. So I think that "enters" is a better word to describe transitions between nodes.

I just think of nodes and transitions as rooms and doors.